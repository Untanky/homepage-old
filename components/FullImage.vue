<template lang="pug">
  //- img.bottom-0(src="/img/profile.jpg")
  .w-full.h-full.bg-center.bg-cover(:style="backgroundImage")
</template>

<script lang="ts">
import Vue, { PropType } from 'vue';
import { FullImage } from '../src/components';

export default Vue.extend({
  name: 'FullImage',
  props: {
    data: {
      type: Object as PropType<FullImage>,
      required: true,
    },
  },
  computed: {
    backgroundImage() {
      return `background-image: url("${this.data.imageUrl}")`;
    },
  },
});
</script>
