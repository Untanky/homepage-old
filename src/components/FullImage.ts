import ComponentData from '../interfaces/ComponentData';

interface FullImage extends ComponentData {
  imageUrl: string;
}

export default FullImage;
