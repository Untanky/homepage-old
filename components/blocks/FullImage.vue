<template lang="pug">
  .w-full.h-full.bg-center.bg-cover(:style="backgroundImage")
</template>

<script lang="ts">
import Vue, { PropType } from 'vue';
import { FullImage } from '../../src/components';

export default Vue.extend({
  name: 'FullImage',
  props: {
    data: {
      type: Object as PropType<FullImage>,
      required: true,
    },
  },
  computed: {
    backgroundImage(): string {
      if (this.data && this.data.imageUrl) {
        return `background-image: url("${this.data.imageUrl}")`;
      }
      return '';
    },
  },
});
</script>
