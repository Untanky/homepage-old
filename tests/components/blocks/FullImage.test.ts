/* eslint-disable no-undef */
// global it, expect
import { mount } from '@vue/test-utils';
import FullImage from '../../../components/blocks/FullImage.vue';

it('should run', () => {
  const props = {
    data: {
      imageUrl: 'https://example.com/image.jpg',
      classes: ['class'],
    },
  };

  const comp = mount(FullImage, { propsData: props });
  expect(comp.element).toMatchSnapshot();
});
