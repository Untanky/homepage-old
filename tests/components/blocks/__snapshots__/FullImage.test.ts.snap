// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should run 1`] = `
<div
  class="w-full h-full bg-center bg-cover"
  style="background-image: url(https://example.com/image.jpg);"
/>
`;
